build-tools/tasks/test-utils.js

```diff
@@ -15,24 +16,132 @@ function toWrapper(componentClass) {
 
 const testUtilsSrcDir = path.resolve('src/test-utils');
 const configs = {
+  common: {
+    // These components are not meant to be present in multiple instances in a single app.
+    // For this reason no findAll and findByTestId finders will be generated for them.
+    noExtraFinders: ['AppLayout', 'TopNavigation'],
+    buildFinder: ({ componentName }) => `
+       ElementWrapper.prototype.find${componentName} = function(selector) {
+         const rootSelector = \`.$\{${toWrapper(componentName)}.rootSelector}\`;
+         // casting to 'any' is needed to avoid this issue with generics
+         // https://github.com/microsoft/TypeScript/issues/29132
+         return (this as any).findComponent(selector ? appendSelector(selector, rootSelector) : rootSelector, ${toWrapper(componentName)});
+       };`,
+    buildExtraFinders: ({ componentName, componentNamePlural }) => `
+      ElementWrapper.prototype.findAll${componentNamePlural} = function(selector) {
+        return this.findAllComponents(${toWrapper(componentName)}, selector);
+      };
+
+      ElementWrapper.prototype.find${componentName}ByTestId = function(testId) {
+        const selector = \`.\${${toWrapper(componentName)}.rootSelector}[data-testid="\${CSS.escape(testId)}"]\`;
+        return this.findComponent(selector, ${toWrapper(componentName)});
+      };`,
+  },
   dom: {
     defaultExport: `export default function wrapper(root: Element = document.body) { if (document && document.body && !document.body.contains(root)) { console.warn('[AwsUi] [test-utils] provided element is not part of the document body, interactions may work incorrectly')}; return new ElementWrapper(root); }`,
-    buildFinderInterface: ({ componentName }) =>
-      `find${componentName}(selector?: string): ${toWrapper(componentName)} | null;`,
+    buildFinderInterface: ({ componentName }) => `
+       /**
+        * Returns the wrapper of the first ${componentName} that matches the specified CSS selector.
+        * If no CSS selector is specified, returns the wrapper of the first ${componentName}.
+        * If no matching ${componentName} is found, returns \`null\`.
+        *
+        * @param {string} [selector] CSS Selector
+        * @returns {${toWrapper(componentName)} | null}
+        */
+       find${componentName}(selector?: string): ${toWrapper(componentName)} | null;`,
+    buildExtraFinderInterfaces: ({ componentName, componentNamePlural }) => `
+       /**
+        * Returns the wrappers of all ${componentNamePlural} that match the specified CSS selector.
```

Comment:
Wording suggestion (not  blocking), I think "an array" is more specific:
```suggestion
        * Returns an array of ${componentNamePlural} wrapper that matches the specified CSS selector.
```

Reply:
That's good suggestion. I just need to double check this later with content 👍